spec/support/tasks.cr

```diff
@@ -24,3 +24,44 @@ class AnotherTask < LuckyCli::Task
   def call
   end
 end
+
+class TaskWithArgs < LuckyCli::Task
+  summary "This task has CLI args"
+  arg :model_name, "This is the name of the model", shortcut: "-m"
+  arg :model_type, description: "Define the model type"
+
+  def call
+    self
+  end
+end
+
+class TaskWithRequiredFormatArgs < LuckyCli::Task
+  summary "This task has a required arg with a format"
+  arg :theme, description: "Specifies which theme to use. Must be dark or light", required: true, format: /^(dark|light)$/
+
+  def call
+    self
+  end
+end
+
+class TaskWithSwitchFlags < LuckyCli::Task
+  summary "This is a task with switch flags"
+
+  switch :force, "Use the force."
+  switch :admin, description: "Set an admin?", shortcut: "-a"
+
+  def call
+    self
+  end
+end
+
+class TaskWithPositionalArgs < LuckyCli::Task
+  summary "This is a task with positional args"
+
+  positional_arg :model, "Define the model", required: true, format: /^[A-Z]/
+  positional_arg :columns, "Define the columns like name:String", to_end: true, format: /\w+:[A-Z]\w+(::\w+)?/
```

Comment:
`positional_arg` and the others don't play well. I would need to know the position of the args that aren't positional to know which index to start at. Or the positional arg would have to know the existence of the others in order to gsub out those..

```crystal
  arg :template, "Add which template you want to use", shortcut: "-t"
  switch :add_auth, "Add in auth", shortcut: "-a"
  positional_arg :options, "Add which options you need", to_end: true, format: /\w+/
```

Right now the `positional_arg` index is 0. That means in this:

`-t BLAH -a one two three`

`template` would be "BLAH", `add_auth` would be true, and `options` would be `["-t BLAH", "-a", "one two three"]`. This would be the case regardless of the order. It's mainly because `arg` and `switch` use the built-in option parser, but `positional_arg` has to hand roll it.

I could do some dirty hacks like tracking if arg or switch are used. In this example it could work, but if you put positonal_arg second, and switch after, but positional_arg said "to_end", it would bork again. Maybe that's just like "hey, you're the one that put grape jelly on your roast beef, you deal with it"..... on a side note, I bet strawberry jam on a turkey sandwich would be bomb... 🤔

Reply:
First off, stop making me hungry 😂 

Second yeah this is complex. Maybe we  just forget it for now and maybe we get ideas come later

This is way better than what we have today so I’m cool with shipping as is!